### PR TITLE
[GEOS-10430] REST API failures

### PR DESCRIPTION
**Pull request by ahmedababnehgeo**
[GEOS-10430](https://osgeo-org.atlassian.net/browse/GEOS-10430)

  

This PR contains a fix for failures in calling /rest/workspaces GET endpoint. Now, the WMS and WMTS store info are both supported.

JIRA ticket is: https://osgeo-org.atlassian.net/browse/GEOS-10430

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [x] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [x] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [x] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [x] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [x] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

File: src/main/src/test/java/org/geoserver/security/decorators/DefaultSecureCatalogFactoryTest.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
+package org.geoserver.security.decorators;
```
Comment (aaime) [inline src/main/src/test/java/org/geoserver/security/decorators/DefaultSecureCatalogFactoryTest.java:1]: Missing copyright header.

File: src/main/src/test/java/org/geoserver/security/decorators/DefaultSecureCatalogFactoryTest.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+import org.easymock.EasyMock;
+import org.geoserver.security.WrapperPolicy;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+public class DefaultSecureCatalogFactoryTest {
```
Comment (aaime) [inline src/main/src/test/java/org/geoserver/security/decorators/DefaultSecureCatalogFactoryTest.java:9]: Test is not checking ``canSecure``

**Review by aaime — CHANGES_REQUESTED** [review]
Looks good, see comments below.

File: src/main/src/test/java/org/geoserver/security/decorators/DefaultSecureCatalogFactoryTest.java (hunk 2 above)
Comment (ahmedababnehgeo) [reply src/main/src/test/java/org/geoserver/security/decorators/DefaultSecureCatalogFactoryTest.java:9]: Good point, I will add a new test for `canSecure` ..

**Comment by ahmedababnehgeo** [comment]
@aaime Tests and copyrights header added

**Review by aaime — APPROVED** [review]

**Review by mprins — APPROVED** [review]

**Comment by mprins** [comment]
as stated in https://github.com/geotools/geotools/pull/3867#issuecomment-1117240834 this is covered by GeoSolutions CLA

**Comment by mprins** [comment]
@ahmedababnehgeo You may want to backport to 2.21.x and 2.20.x branches

**Comment by jodygarnett** [comment]
I directly cherry-picked to 2.21.x and 2.20.x (since we are cutting release candidate and rounding up fixes)

**Comment by ahmedababnehgeo** [comment]
Thanks everyone for feedback ...

**Pull request merged**